pandas/tests/io/test_sql.py

```diff
@@ -1430,6 +1430,28 @@ def test_sqlalchemy_integer_overload_mapping(self, integer):
         ):
             sql.SQLTable("test_type", db, frame=df)
 
+    def test_sqlalchemy_oracle_number_type_mapping(self):
+        from numpy import (
+            float64,
+            int64,
+        )
+        from sqlalchemy.dialects.oracle import NUMBER
+
+        # GH34988 Lost precision of NUMBER in read_sql_table()
+        df = DataFrame([0, 1], columns=["a"], dtype=float64)
+        db = sql.SQLDatabase(self.conn)
+        table = sql.SQLTable("test_type", db, frame=df)
+
+        column = sqlalchemy.sql.schema.Column(
+            "a",
+            NUMBER(precision=5, scale=2),
+        )
+        col_type = table._get_dtype(column.type)
```

Comment:
Is there a way to test this from the end user perspective? Typically we don't want to reach into internals like `_get_dtype` when setting up tests; would be best to simulate what users would do

Reply:
I'm not sure I can think of one without standing up an Oracle instance into the test runner. Given that the "NUMBER" type is only particular to the Oracle dialect, it makes it quite hard to test. I did look at potentially doing it through the oracle dialect in sqlalchemy itself, however it still causes an issue in pandas without this change. Would appreciate some insight into another approach if you can think of one?

Reply:
Hmm OK. Can you model this test closer to what is being done in `test_sqlalchemy_integer_mapping` ?